### PR TITLE
Fixed write_after_write log message in sync val

### PR DESCRIPTION
**Pull request by locke-lunarg**

### CODE REVIEW

File: layers/synchronization_validation.cpp [hunk 1]
```diff
@@ -146,7 +146,7 @@ static std::string string_SyncStageAccessFlags(SyncStageAccessFlags flags, const
 
 static std::string string_UsageTag(const HazardResult &hazard) {
     const auto &tag = hazard.tag;
-    assert(hazard.usage_index < syncStageAccessInfoByStageAccessIndex.size());
+    assert(static_cast<size_t>(hazard.usage_index) < syncStageAccessInfoByStageAccessIndex.size());
```
Comment (jzulauf-lunarg) [inline layers/synchronization_validation.cpp:149]: Why this change? usage_index 0 is valid (SYNC_DRAW_INDIRECT_INDIRECT_COMMAND_READ)

If you're concerned that this is a valid hazard, test hazard.hazard != NONE, but the original is an out of range check I'd like to retain.

Comment (locke-lunarg) [reply layers/synchronization_validation.cpp:149]: Because in some devices, the compile fails because  the two data type are different and wrong, I just tried making both sides are the same type.

**Comment by jzulauf-lunarg** [comment]
The assert means that HazardResult isn't valid... perhaps I'm stuffing a SyncStageAccessFlags into it, where it should be a ...Index?  or something uninitialized?
